shards/core/brancher.hpp

```diff
@@ -41,38 +45,43 @@ struct Brancher {
 
   const shards::ExposedInfo &getMergedRequirements() const { return _mergedRequirements; }
 
-  void compose(const SHInstanceData &data) {
+  void compose(const SHInstanceData &data, const IgnoredVariables &ignored = {}) {
     _collectedRequirements.clear();
 
     for (auto &wire : wires) {
       composeSubWire(data, wire);
     }
 
     // Merge requirements from compose result
-    for (auto &wire : wires) {
-      for (auto &req : wire->composeResult->requiredInfo) {
-        _mergedRequirements.push_back(req);
-      }
-    }
+    // TODO: Remove since this is already checked by _collectedRequirements below
+    // for (auto &wire : wires) {
+    //   for (auto &req : wire->composeResult->requiredInfo) {
+    //     _mergedRequirements.push_back(req);
+    //   }
+    // }
 
     // Merge deep requirements
     if (captureAll) {
       for (auto &avail : data.shared) {
+        if (ignored.find(avail.name) != ignored.end())
+          continue;
         SHLOG_TRACE("Branch: adding variable to requirements: {}", avail.name);
         _mergedRequirements.push_back(avail);
       }
     } else {
       for (auto &avail : data.shared) {
         auto it = _collectedRequirements.find(avail.name);
         if (it != _collectedRequirements.end()) {
+          if (ignored.find(avail.name) != ignored.end())
```

Comment:
What's this ignored?

Reply:
Values that shouldn't be exposed inside the branch, although I think I don't need it anymore as I'm filtering object variables out of the capturing brancher now

Reply:
yeah adds complexity